drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -1070,7 +1072,11 @@ void nrf_802154_receive_failed(nrf_802154_rx_error_t error, uint32_t id)
 
 #if defined(CONFIG_IEEE802154_CSL_ENDPOINT)
 	if (id == DRX_SLOT_RX && error == NRF_802154_RX_ERROR_DELAYED_TIMEOUT) {
-		return;
+		if (nrf5_data.event_handler && nrf5_data.rx_on_when_idle) {
```

Comment:
It should return when `nrf5_data.rx_on_when_idle=off` regardless of the value of `nrf5_data.event_handler`.